### PR TITLE
🌱 Enable ginkgolinter len assertion check and fix findings

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

Enable ginkgolinter len assertion check and fix findings. 

A minor improvement imo, but that still an improvement. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partof #8050

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/lgtm

File: exp/runtime/internal/controllers/extensionconfig_controller_test.go [hunk 1]
```diff
@@ -261,12 +261,12 @@ func TestExtensionReconciler_discoverExtensionConfig(t *testing.T) {
 
 		// Expect exactly one handler and expect the name to be the handler name plus the extension name.
 		handlers := discoveredExtensionConfig.Status.Handlers
-		g.Expect(len(handlers)).To(Equal(0))
+		g.Expect(handlers).To(HaveLen(0))
```
Comment (elmiko) [inline exp/runtime/internal/controllers/extensionconfig_controller_test.go:264]: should we use `BeEmpty()` here?

**Review by elmiko — COMMENTED** [review]
looks mostly good to me, just a question about consistency

File: exp/runtime/internal/controllers/extensionconfig_controller_test.go (hunk 1 above)
Comment (oscr) [reply exp/runtime/internal/controllers/extensionconfig_controller_test.go]: @elmiko Great find! Neither I nor the linter picked up on this. But I think you are absolutely right and it's fixed everywhere.

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**